### ISSUE TITLE
Suspicious activity detected

### ISSUE THREAD

**Issue by tomb56** [KEN-2000l/EducationPerfected#36]
i've been using v2 for a while and it has been working well but i tryed to run it a couple of minutes ago and it says Suspicious activity detected. I think there's been a update to EP because when i ran the normal version it sayed update in the bottom left corner.

**Comment by tomb56** [comment]
i don't even have to run the script and it does it it just does it after about 2 seconds
![SharedScreenshot](https://user-images.githubusercontent.com/98370757/168732260-699a08ba-51a4-43af-a2e7-26b393666657.jpg)

**Comment by KEN-2000l** [comment]
bruh... that was quick

stg if they implemented like a browser check...
I'll look at it later

**Comment by tomb56** [comment]
i just had japanese and was using it for some work that i got set, i did it legit anyway so dont need. Just letting you know

**Comment by KEN-2000l** [comment]
i literally dk why
it just happens when I click the start button of a task, even though absolutely no functions are triggered D:

**Comment by tomb56** [comment]
Yeah it weird

**Comment by tomb56** [comment]
Yeah it weird idk how it could detect it if it does do anything

**Comment by tomb56** [comment]
Yeah it weird idk how it could detect it if it doesn't do anything

**Comment by tomb56** [comment]
Yeah it weird idk how it could detect it if it doesn't do anything

**Comment by Susmaster64** [comment]
EP is tracking repo?

**Comment by tomb56** [comment]
when do you recon you'll have it fixed by

**Comment by Susmaster64** [comment]
i have a python script that might work with semi auto mode but its slower but it does what fully automatic does, but it requires some libraries to be installed. I can link the repo. It has random inputting delay and typing time.

**Comment by Susmaster64** [comment]
maybe it can detect because ep can detect that there is puppeteer?

**Comment by Razzy52** [comment]
It comes up with that if you get too many right in a row, and too quickly {I think} You'll need to do like 2 wrong for every one right.

**Comment by keyp0s** [comment]
has anyone tried using other webdrivers like selenium. or have they just blocked all developer browsers

**Comment by KEN-2000l** [comment]
i think it is a ratelimit thing
where similar to what razzy said, if you get like 10 correct in a second the anticheat will trigger after a while

**Comment by KEN-2000l** [comment]
this is problematic as it breaks one of the essential features of the script with (almost) no way to fix it; the high-speed answering

it's hard to fix because it is a server-side rate limit afaik... only chance of bypassing is messing with packets/networking but I doubt that it would work (and if there is a viable exploit, can always be fixed by them)

**Comment by keyp0s** [comment]
> i think it is a ratelimit thing where similar to what razzy said, if you get like 10 correct in a second the anticheat will trigger after a while

yea just tried it again and seems to be working. they definitely must have some sort of rate limiting. you could always just run it slower but over like a whole day. but as you were saying before this is problematic as it breaks the essential feature of the high speed answering

**Comment by KEN-2000l** [comment]
it is pretty confusing:

- i have just used it at full speed for quite a while with no problems
- last time, I get the anticheat popup when clicking start on a task on the puppeteer window; I could log in from my personal browser and do tasks with no problems

**Comment by KEN-2000l** [comment]
yup, similar experience
still not sure what exactly triggers the anticheat

**Comment by suryasingh1** [comment]
Umm i think it is the submit button cause i used an ahk file to press enter instead and it works completely fine. Or it might detect any function triggers through injected scripts like button press or anything.

**Comment by keyp0s** [comment]
yea i had that exact same experience. as soon as the script entered the submit button i was kicked from the task for suspicious behaviour.

**Comment by suryasingh1** [comment]
Do you guys want me to send my ahk files. What i have done is copied another guys script to make it type like a human. But it goes like this what between 2 to 3 seconds. Type like human press enter. The commands are Carl+t to start Carl+r to stop

**Comment by KEN-2000l** [comment]
one thing is that like even if I don't activate the script it has triggered the anticheat
but nothing has consistent reproduction results D:

**Comment by suryasingh1** [comment]
Isn’t that only for the new version?

**Comment by KEN-2000l** [comment]
oh were you talking about the old userscript?
well then yeah they detect it when you call `.click()` on the submit button

**Comment by chinakillertianamen** [comment]
> Do you guys want me to send my ahk files. What i have done is copied another guys script to make it type like a human. But it goes like this what between 2 to 3 seconds. Type like human press enter. The commands are Carl+t to start Carl+r to stop

can you send the ahk files I want to run some tests with it.

**Comment by chinakillertianamen** [comment]
dont mind my username

**Comment by akioweh** [comment]
@suryasingh1 :) (in case they don't have notifications for this thread)

**Comment by suryasingh1** [comment]
sorry for very late reply here.

Github doesn't support the file type so i will send in text. Here is the first script:

#NoEnv  ; Recommended for performance and compatibility with future AutoHotkey releases.
; #Warn  ; Enable warnings to assist with detecting common errors.
SendMode Input  ; Recommended for new scripts due to its superior speed and reliability.
SetWorkingDir %A_ScriptDir%  ; Ensures a consistent starting directory.

DetectHiddenWindows, On
SetTitleMatchMode, 2

WinGetClass, WinClass, Google Chrome

loop {
SetTimer, GetFocus, 200								; Makes the browser gain focus every 200 ms
}
return

GetFocus:
ControlFocus,, Google Chrome
Return

setkeydelay 300
looping := true
^t::
looping := true
While(looping = true)
{
Random, rand, 1100, 1200
Sleep %rand%
RunWait, HumanTyping.ahk
Sleep 150
Send, {Enter}
}
return 
^r::
looping := false
return

**Comment by suryasingh1** [comment]
here is the second one (HumanTyping.ahk): 

#NoEnv  ; Recommended for performance and compatibility with future AutoHotkey releases.
; #Warn  ; Enable warnings to assist with detecting common errors.
SendMode Input  ; Recommended for new scripts due to its superior speed and reliability.
SetWorkingDir %A_ScriptDir%  ; Ensures a consistent starting directory.


AutoTrim,On
string = %clipboard%
Gosub,ONLYTYPINGCHARS
StringSplit, charArray, string
Loop %charArray0%
{
	this_char := charArray%a_index%
	Send %this_char%
	Random, typeSlow, 1, 3
	typeMin = 150
	typeMax = 200
	if (typeSlow >= 3){
		typeMin = 200
		typeMax = 250
	}
	Random, t, typeMin, typeMax
	Sleep, %t%
}
Return

ONLYTYPINGCHARS:
AutoTrim,Off
StringCaseSense,On
StringReplace,string,string,–,-,All	;emdash
StringReplace,string,string,´,',All
StringReplace,string,string,’,',All
StringReplace,string,string,©,(C),All
StringReplace,string,string,“,",All	;left quote
StringReplace,string,string,”,",All	;right quote
StringReplace,string,string,®,(R),All
StringReplace,string,string,¼,1/4,All
StringReplace,string,string,½,1/2,All
StringReplace,string,string,¾,3/4,All
StringReplace,string,string,™,TM,All
StringReplace,string,string,«,<<,All
StringReplace,string,string,»,>>,All
StringReplace,string,string,„,',All
StringReplace,string,string,•,-,All	;bullet
StringReplace,string,string,…,...,All
StringReplace,string,string,`r`n,`n,All ;replace newlines
StringReplace,string,string,chr(0),A_Space,All ;NULL
StringReplace,string,string,chr(9),A_Space,All ;Horizontal Tab
StringReplace,string,string,chr(10),A_Space,All ;Line Feed
StringReplace,string,string,chr(11),A_Space,All ;Vertical Tab
StringReplace,string,string,chr(14),A_Space,All ;Column Break
StringReplace,string,string,chr(160),A_Space,All ;Non-breaking space
Return

**Comment by qmanstal** [comment]
> here is the second one (HumanTyping.ahk):
> 
> #NoEnv ; Recommended for performance and compatibility with future AutoHotkey releases. ; #Warn ; Enable warnings to assist with detecting common errors. SendMode Input ; Recommended for new scripts due to its superior speed and reliability. SetWorkingDir %A_ScriptDir% ; Ensures a consistent starting directory.
> 
> AutoTrim,On string = %clipboard% Gosub,ONLYTYPINGCHARS StringSplit, charArray, string Loop %charArray0% { this_char := charArray%a_index% Send %this_char% Random, typeSlow, 1, 3 typeMin = 150 typeMax = 200 if (typeSlow >= 3){ typeMin = 200 typeMax = 250 } Random, t, typeMin, typeMax Sleep, %t% } Return
> 
> ONLYTYPINGCHARS: AutoTrim,Off StringCaseSense,On StringReplace,string,string,–,-,All ;emdash StringReplace,string,string,´,',All StringReplace,string,string,’,',All StringReplace,string,string,©,(C),All StringReplace,string,string,“,",All ;left quote StringReplace,string,string,”,",All ;right quote StringReplace,string,string,®,(R),All StringReplace,string,string,¼,1/4,All StringReplace,string,string,½,1/2,All StringReplace,string,string,¾,3/4,All StringReplace,string,string,™,TM,All StringReplace,string,string,«,<<,All StringReplace,string,string,»,>>,All StringReplace,string,string,„,',All StringReplace,string,string,•,-,All ;bullet StringReplace,string,string,…,...,All StringReplace,string,string,`r`n,`n,All ;replace newlines StringReplace,string,string,chr(0),A_Space,All ;NULL StringReplace,string,string,chr(9),A_Space,All ;Horizontal Tab StringReplace,string,string,chr(10),A_Space,All ;Line Feed StringReplace,string,string,chr(11),A_Space,All ;Vertical Tab StringReplace,string,string,chr(14),A_Space,All ;Column Break StringReplace,string,string,chr(160),A_Space,All ;Non-breaking space Return

is this an alternative?

**Comment by Susmaster64** [comment]
I just reccomend using the old version to copy the text to clipboard and use my  little clipboard to typing python script, it's here somewhere on one of the issues, and I think the repo is under my account somewhere.
